chain-signatures/node/src/protocol/signature.rs

```diff
@@ -621,13 +612,14 @@ impl SignatureManager {
                             payload_hash: generator.request.payload.into(),
                         };
                         if generator.proposer == self.me {
-                            self.signatures.push(ToPublish::new(
+                            let to_publish = ToPublish::new(
                                 sign_request_id.request_id,
                                 request,
                                 generator.sign_request_timestamp,
                                 output,
                                 generator.request.chain,
-                            ));
+                            );
+                            rpc.publish(self.public_key, to_publish).await;
```

Comment:
nit: this await might block the loop if the channel used inside `rpc.publish` is full. If there are circular dependencies between channels and/or locks, it might even lead to a deadlock.

Perhaps spawning the future is safer. But it should only matter in edge cases.